### PR TITLE
Add some async/clock utilities

### PR DESCRIPTION
**Pull request by mwachs5**
This provides some common "utilities" (black box hacks) to do some clock/reset related things that aren't possible in Chisel.

### CODE REVIEW

**Comment by aswaterman** [comment]
thanks!

File: vsim/Makefrag [hunk 1]
```diff
@@ -9,6 +9,9 @@ bb_vsrcs = $(base_dir)/vsrc/DebugTransportModuleJtag.v \
 	$(base_dir)/vsrc/AsyncMailbox.v \
 	$(base_dir)/vsrc/AsyncResetReg.v \
 	$(base_dir)/vsrc/ClockDivider.v \
+        $(base_dir)/vsrc/ClockToSignal.v \
+        $(base_dir)/vsrc/SignalToClock.v \
```
Comment (yunsup) [inline vsim/Makefrag]: Please use tabs within a Makefile

**Comment by zhemao** [comment]
I think this is something we should just add to Chisel.

File: vsim/Makefrag (hunk 1 above)
Comment (mwachs5) [inline vsim/Makefrag]: sorry...  need to set emacs to recognize "Makefrag" as makefile-mode.

**Comment by mwachs5** [comment]
@zhemao absolutely. But I got tired of not having it at all, so this is the current workaround.

**Pull request merged**